### PR TITLE
Joel (freedom branch. freedoooooooom)

### PR DESCRIPTION
**Pull request by Joelzor**

### CODE REVIEW

File: src/client/App.jsx [hunk 1]
```diff
@@ -1,56 +1,174 @@
...
-import './App.css';
-import MovieForm from './components/MovieForm';
-import UserForm from './components/UserForm';
+import { useState, useEffect } from "react";
+import "./App.css";
+import UserForm from "./components/UserForm";
+import MovieForm from "./components/MovieForm";
 
-const apiUrl = 'http://localhost:4000';
+const apiUrl = "http://localhost:4000";
```
Comment (fatimaahmed14) [inline src/client/App.jsx:7]: props to you for using the freedom branch, wouldn't expect any less from you

File: src/client/App.jsx [hunk 2]
```diff
@@ -1,56 +1,174 @@
...
+        <>
+          <p style={{ color: "lightgreen" }}>{registerResponse}</p>
+          {removeAlert()}
+        </>
+      )}
+
+      <h2>Log in</h2>
+      <UserForm handleSubmit={loginUser} />
+
+      {loginError && (
```
Comment (fatimaahmed14) [inline src/client/App.jsx:136]: this is so cool, you have it change to red when there's an error???!!!

File: src/server/controllers/user.js [hunk 3]
```diff
@@ -1,39 +1,69 @@
...
-    const foundUser = null;
+        res.status(201).json({ status: "success", user: newUser });
+      } catch (error) {
+        if (error.code === "P2002") {
+          res
+            .status(403)
+            .json({ error: `The username ${username} is already taken` });
+        } else {
+          res.status(500).json({ error });
+        }
```
Comment (fatimaahmed14) [inline src/server/controllers/user.js:36]: error handling is wonderfully done

Comment (Joelzor) [reply src/server/controllers/user.js:36]: I mean this part is just what Julian showed us, I would love to take credit though ; )

**Pull request closed without merging**